src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration.java

```diff
@@ -55,17 +59,59 @@ public GitHubConfiguration() {
     }
 
     @NonNull
-    public List<Endpoint> getEndpoints() {
+    public synchronized List<Endpoint> getEndpoints() {
```

Comment:
🐛 I doubt it will work correctly since you return the original collection outside. `unmodifiableList ` does not do a copy of the object, hence it will be a subject for concurrency issues. I would rather switch on CopyOnWrite collection